core/node/libp2p/rcmgr.go

```diff
@@ -218,6 +238,122 @@ func NetStat(mgr network.ResourceManager, scope string) (NetStatOut, error) {
 	}
 }
 
+var scopes = []string{
+	config.ResourceMgrSystemScope,
+	config.ResourceMgrTransientScope,
+	config.ResourceMgrServiceScopePrefix,
+	config.ResourceMgrProtocolScopePrefix,
+	config.ResourceMgrPeerScopePrefix,
+}
+
+func limitToScope(l rcmgr.BaseLimit) *network.ScopeStat {
+	return &network.ScopeStat{
+		NumStreamsInbound:  l.StreamsInbound,
+		NumStreamsOutbound: l.StreamsOutbound,
+		NumConnsInbound:    l.ConnsInbound,
+		NumConnsOutbound:   l.ConnsOutbound,
+		NumFD:              l.FD,
+		Memory:             l.Memory,
+	}
+}
+
+// compareScopes copares stat and limit.
+// If any of the stats value are equals or above the specified percentage,
+// stat object is returned.
+func compareScopes(stat, limit *network.ScopeStat, percentage int) *network.ScopeStat {
+	if stat == nil || limit == nil {
+		return nil
+	}
+	if abovePercentage(int(stat.Memory), int(limit.Memory), percentage) {
+		return stat
+	}
+	if abovePercentage(stat.NumConnsInbound, limit.NumConnsInbound, percentage) {
+		return stat
+	}
+	if abovePercentage(stat.NumConnsOutbound, limit.NumConnsOutbound, percentage) {
+		return stat
+	}
+	if abovePercentage(stat.NumFD, limit.NumFD, percentage) {
+		return stat
+	}
+	if abovePercentage(stat.NumStreamsInbound, limit.NumStreamsInbound, percentage) {
+		return stat
+	}
+	if abovePercentage(stat.NumStreamsOutbound, limit.NumStreamsOutbound, percentage) {
+		return stat
+	}
+
+	return nil
+}
+
+func abovePercentage(v1, v2, percentage int) bool {
+	if percentage == 0 {
+		return true
+	}
+
+	if v2 == 0 {
+		return false
+	}
+
+	return int((v1/v2))*100 >= percentage
+}
+
+func NetLimitAll(mgr network.ResourceManager) (*NetStatOut, error) {
```

Comment:
IMO the output of this should be consistent w/ the stats, can we use BaseLimit for the NetStatOut values instead of ScopeStat?

Example, IMO these should all have exactly the same schema:

$ ipfs swarm limit system | jq .
{
  "Conns": 1800,
  "ConnsInbound": 64,
  "ConnsOutbound": 4611686018427388000,
  "FD": 256,
  "Memory": 134217728,
  "Streams": 4611686018427388000,
  "StreamsInbound": 4611686018427388000,
  "StreamsOutbound": 4611686018427388000
}
$ ipfs swarm stats all | jq .System
{
  "Memory": 3706880,
  "NumConnsInbound": 9,
  "NumConnsOutbound": 77,
  "NumFD": 27,
  "NumStreamsInbound": 21,
  "NumStreamsOutbound": 62
}
$ ipfs swarm limit all | jq .System
{
  "Memory": 134217728,
  "NumConnsInbound": 64,
  "NumConnsOutbound": 4611686018427388000,
  "NumFD": 256,
  "NumStreamsInbound": 4611686018427388000,
  "NumStreamsOutbound": 4611686018427388000
}

Reply:
Actual output:

```
➜  ./cmd/ipfs/ipfs swarm limit system | jq .
{
  "Conns": 1800,
  "ConnsInbound": 314,
  "ConnsOutbound": 4611686018427388000,
  "FD": 524288,
  "Memory": 4331667456,
  "Streams": 4611686018427388000,
  "StreamsInbound": 5027,
  "StreamsOutbound": 4611686018427388000
}

➜  ./cmd/ipfs/ipfs swarm stats all | jq .System
{
  "Conns": 210,
  "ConnsInbound": 0,
  "ConnsOutbound": 210,
  "FD": 61,
  "Memory": 14974976,
  "Streams": 218,
  "StreamsInbound": 17,
  "StreamsOutbound": 201
}

➜  ./cmd/ipfs/ipfs swarm limit all | jq .System
{
  "Conns": 1800,
  "ConnsInbound": 314,
  "ConnsOutbound": 4611686018427388000,
  "FD": 524288,
  "Memory": 4331667456,
  "Streams": 4611686018427388000,
  "StreamsInbound": 5027,
  "StreamsOutbound": 4611686018427388000
}
```